src/components/FilterComponent/FilterDropdown.jsx

```diff
@@ -0,0 +1,228 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import DatePicker from './DatePicker';
+
+export const success = 'success';
+export const failure = 'failure';
+export const onboarding = 'onboarding';
+export const offboarding = 'offboarding';
+export const partnerName = 'partner';
+export const fellowName = 'fellow';
+export const emailAutomations = 'emailAutomation';
+export const slackAutomations = 'slackAutomation';
+export const freckleAutomations = 'freckleAutomation';
+
+/**
+ * Displays a custom checkbox
+ */
+export const Checkbox = ({
+  checkCondition, id, label, name, noLabel = false, ...rest
+}) => (noLabel ? (
+  <React.Fragment>
+    <input
+      type="checkbox"
+      id={id}
+      name={name}
+      checked={checkCondition}
+      {...rest}
+    />
+    <span className="checkmark" />
+  </React.Fragment>
+) : (
+  <label className={`${id}-label`} htmlFor={id}>
+    <input
+      type="checkbox"
+      id={id}
+      name={name}
+      checked={checkCondition}
+      {...rest}
+    />
+    <span className="checkmark" />
+    {label}
+  </label>
+)
+);
+
+Checkbox.defaultProps = {
+  noLabel: false,
+  label: '',
+};
+Checkbox.propTypes = {
+  checkCondition: PropTypes.bool.isRequired,
+  id: PropTypes.string.isRequired,
+  name: PropTypes.string.isRequired,
+  label: PropTypes.string,
+  noLabel: PropTypes.bool,
+};
+
+
+const FilterDropdown = (props) => {
+  const {
+    filters: {
+      searchBy, type, emailAutomation, slackAutomation, freckleAutomation, date,
+    },
+    handleInputChange,
+  } = props;
+
+  const handleDateChange = (selectedDate, selection) => {
+    const e = { target: { name: 'date', id: selection, value: selectedDate }, persist: () => {} };
+    handleInputChange(e);
+  };
+
+  return (
+    <div className="filter-dropdown-parent">
+      <div className="search-by">
+        <p className="search-title"><b>Search by</b></p>
+        <Checkbox
+          checkCondition={searchBy === fellowName}
```

Comment:
there are a lot of repetitions going on here, you need to refactor the code to be DRY